packages/ruleset/src/utils/date-based-utils.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2024 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+/**
+ * This function takes a schema property name and uses a collection of regular
+ * expressions to heuristically determine if the property is a date or a date
+ * time property. It returns `true` if the name matches one of the expressions.
+ *
+ * @param {string} name the name of a schema property
+ * @returns a boolean value indicating that the property seems to be date-based
+ */
+function isDateBasedName(name) {
+  // Check for matching against certain patterns.
+  const dateBasedNamePatterns = [
+    // Any name ending in `_at`.
+    new RegExp('.*_at$'),
+
+    // Any name ending in `_on`.
+    new RegExp('.*_on$'),
+
+    // Any name starting with `date_`.
+    new RegExp('^date_.*'),
+
+    // Any name containing `_date_`.
+    new RegExp('.*_date_.*'),
+
+    // Any name ending in `_date`.
+    new RegExp('.*_date$'),
+
+    // Any name starting with `time_`.
+    new RegExp('^time_.*'),
+
+    // Any name containing `_time_`.
+    new RegExp('.*_time_.*'),
+
+    // Note: not including any name ending in `_time` because it was too easy
+    // to think of counterexamples. `running_time` in the "movies" API of our
+    // test document in this project is one of them.
+
+    // Any name containing `timestamp`.
+    new RegExp('.*timestamp.*'),
+  ];
+
+  if (dateBasedNamePatterns.some(regex => regex.test(name))) {
+    return true;
+  }
+
+  // Check for direct equality with certain terms.
+  const dateBasedNames = [
+    'created',
```

Comment:
You could just fold these into the array of regexp's above (e.g. `new RegExp('^created$')` with proper anchoring to ensure the direct equality match.  That way, you have just one set of regex's to check.

Reply:
Good idea! 👍

Reply:
Fixed